imsim/camera.py

```diff
@@ -183,3 +221,21 @@ def update(self, other):
     def __getattr__(self, attr):
         """Provide access to the attributes of the underlying lsst_camera."""
         return getattr(self.lsst_camera, attr)
+
+
+def CcdSaturation(config, base, value_type):
+    """Retrieve detector data from the Camera object."""
+    camera_object = GetInputObj('camera_data', config, base, 'CcdSaturation')
+
+    req = {'det_name': str}
+
+    kwargs, safe = GetAllParams(config, base, req=req)
+    det_name = kwargs['det_name']
+
+    value = camera_object[det_name].full_well
+    return value, False
+
+
+RegisterInputType('camera_data', InputLoader(Camera, takes_logger=False))
```

Comment:
I think the parameters for this new camera_data type aren't being checked anywhere with a Get or Check parameters call.

Reply:
I think they must be used inside of the galsim code when it provides the `CcdSaturation` value to the diffraction_psf object.

Reply:
@cwwalter The `_req_params` and `_opt_params` class variables mean that they get checked.

Reply:
Ah from `get_cls_params` in the inputLoader. Thanks.